pkgs/development/mobile/androidenv/README.md

```diff
@@ -6,7 +6,7 @@
 4. Update the relevant argument defaults in `compose-android-packages.nix`
 
 # How to run tests
-You may need to make yourself familiar with [tests](https://nixos.org/manual/nixpkgs/stable/#var-meta-tests), and [Writing larger package tests](https://nixos.org/manual/nixpkgs/stable/#ssec-package-tests-writing) in the Manual, then run tests locally with:
+You may need to make yourself familiar with [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests), and [Writing larger package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#writing-larger-package-tests), then run tests locally with:
```

Comment:
Those kind of links are brittle. Because it doesn't link to a specific commit it can happen that somebody moves or renames the heading in that file and will break the link without anyone noticing.

Reply:
Thanks for poking at this. Forgot to leave a comment on it.

Roughly intending these two link changes as a query to @infinisil about what internal READMEs should be doing here. 

- The first link could technically keep pointing to the manual, but the existing links from the repo to the stable manual basically mean that I can't update the underlying anchors that the README links use without breaking them until the stable manual catches up. 

    (Maybe it's fine to change it to change it to the unstable manual. Maybe it's fine to just let the legacy anchor catch these, but it did feel like I should go ahead and update them. In the specific context of _this_ README, it felt like sending them to the internal contributor docs in both cases is probably a small improvement.)
- I agree the links can be fragile, but the `ssec-package-tests-writing` section was roughly hollowed out and replaced with equivalent links into the master branch of the repo in the course of #245243, and this is just removing a click from the reader's path to reach the information they need:

    https://github.com/NixOS/nixpkgs/pull/245243/files#diff-99f4eb67c3def9d78bbb9ec86a7dced190f46ee7151c77dcbb15a6a08727a330L565-R39
- I guess I could technically make these relative file links so that it'd be easier to notice that they're broken during work on some other branch. (I'm skeptical it would often get noticed unless we have tooling to check for it.)

    ```suggestion
    You may need to make yourself familiar with [package tests](../../../README.md#package-tests), and [Writing larger package tests](../../../README.md#writing-larger-package-tests), then run tests locally with:
    ```
    
    I'll save actually applying that fix until infinisil weighs in, though.

Reply:
Some notes:
- Fully agree that it's a problem that markdown links could become invalid. We should use https://github.com/serokell/xrefcheck to fix this, would be super easy to run in CI actually
- Fully agree that it's a problem that anchors could become invalid when the sections are renamed. There's no good fix here, other than trying not to or to manually insert backwards compatible aliases.
- The proper fix really is to create some infrastructure for building all Markdown files throughout Nixpkgs into a separate contributors manual, with links between them, some custom syntax for stable anchors, etc.
- Yes at that point it's almost like the user reference manual, but with notably differences (see also https://github.com/NixOS/nixpkgs/issues/244056):
  - There should only be one rendered version of the contributors manual, there's no "stable" contributor docs, because you can't contribute to a past version
  - We can make "breaking" changes to the contributor docs at any time
  - The contributor docs aren't useful to a non-contributor
  - The contributor docs should mirror the repo structure (which itself also isn't stable) and be easily discoverable from that.

And the relative links are definitely the way to go imo, because they also work on different branches :)